### ISSUE TITLE
A badge with no image extension is assumed to be PNG

### ISSUE THREAD

**Issue by maelle** [quarto-dev/quarto-cli#6092]
### Bug description

In a blog where one of the post has this badge

```
[![OpenSSF Best Practices](https://bestpractices.coreinfrastructure.org/projects/1882/badge)](https://bestpractices.coreinfrastructure.org/projects/1882)
```

the badge is converted to

```
<p><a href="https://bestpractices.coreinfrastructure.org/projects/1882"><img src="https://bestpractices.coreinfrastructure.org/projects/1882/badge.png" class="img-fluid figure-img"></a></p> 
```

therefore the image is missing.

Interestingly in the post listing (where I would not expect it to be included, other issue #6091), the badge is

```
<a href="./posts/post-with-code/index.html"> <img src="https://bestpractices.coreinfrastructure.org/projects/1882/badge" class="thumnail-image"> </a>
```

It might unusual for a badge to be specified without extension, though. I see it's also done for metacran badges https://github.com/r-lib/usethis/blob/d3614c701b9e2ee6f680a245d3315b1b941f49de/README.Rmd#L21

### Steps to reproduce

https://github.com/maelle/badgeblog/blob/main/posts/post-with-code/index.qmd#L8

### Expected behavior

I would expect no extension to be added to the image in the badge.

### Actual behavior

A PNG extension is added.

### Your environment

RStudio 2023.03.0+386 "Cherry Blossom" Release (3c53477afb13ab959aeb5b34df1f10c237b256c3, 2023-03-09) for Ubuntu Bionic
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) rstudio/2023.03.0+386 Chrome/108.0.5359.179 Electron/22.0.3 Safari/537.36

### Quarto check output

```bash
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 2.19.2: OK
      Dart Sass version 1.32.8: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.49
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.8.10
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.2.0
      Path: /opt/R/4.2.0/lib/R
      LibPaths:
        - /home/maelle/R/x86_64-pc-linux-gnu-library/4.2
        - /opt/R/4.2.0/lib/R/library
      rmarkdown: 2.22

[✓] Checking Knitr engine render......OK
```

**Comment by cscheid** [comment]
The problem here is that we use `default-image-extension: png` by default as a reader option.

**Comment by maelle** [comment]
thank you!!

**Comment by brunj7** [comment]
I add a similar problem (with `![CERN computing center](https://cds.cern.ch/images/OPEN-PHO-CCC-2017-001-1/file?size=medium)` not rendering.  I am glad found this GitHub issue, this solved my problem as well, thank you!!

My feedback  (take it or leave it 🙂, especially since I do not know the motivation behind this choice) would be to encourage you to reconsider this default behavior. IMHO the image should break if the provided path is incomplete / missing the extension. This would be an easier thing to debug than an URL "automatically" being altered.

**Comment by cscheid** [comment]
I agree that this default behavior is bad - if we're going to change this, then we should consider the change carefully and at the beginning of a release cycle.

**Comment by jjallaire** [comment]
This is there to accommodate multiple format workflows (so the same source can target a .png for web output and a .pdf for LaTeX output). I believe we have a lot of users already relying on this behavior so I don't think we should change the default. We could however issue a warning in the case where there isn't more than one image to resolve. We could also implement the feature directly in Lua and fall back to not appending the extension if there is no target file while there _is a target_ file for the variation without an extension.

**Comment by matteopedani** [comment]
default-image-extension: "" 

My site have all images in google drive. Where  put default-image-extension: ""   for all pages in a site, and not in a single page? I have 200 pages.

**Comment by mcanouil** [comment]
In your project level configuration file, or in the directory level configuration file or in the frontmatter, see <https://quarto.org/docs/projects/quarto-projects.html#shared-metadata>.

**Comment by mcanouil** [comment]
FYI, another issue with the default extension enforcement is that it means you cannot use:

```markdown
![An iframe/embed because the link is a website](https://example.com/)
```

Because Quarto assumes `![]()` is only images it breaks Pandoc iframe feature.

```html
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://example.com/.png" class="img-fluid figure-img"></p>
<figcaption>An iframe/embed because the link is a website</figcaption>
</figure>
</div>
```

Current workaround is to add `index.html` or anything with a file extension.

```markdown
![An iframe/embed because the link is a website](https://example.com/index.html)
```

```html
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><embed src="https://example.com/index.html" class="img-fluid"></p>
<figcaption>An iframe/embed because the link is a website</figcaption>
</figure>
</div>
```

**Comment by cderv** [comment]
@mcanouil curious about this usage: `![An iframe/embed because the link is a website](https://example.com/)`

Is this to leverage Pandoc converting unknown image to `embed` but still have figure node and caption ? 

It feels a bit weird to me to use a html link inside an image markdown syntax. So I am probably missing something here. 

Thanks

**Comment by mcanouil** [comment]
Literally to embed/iframe content such as PDF and Reveal.js slides or possibly any content really.

```markdown
![Slides](https://quarto.org/docs/presentations/revealjs/demo/index.html)

![Mickaël CANOUIL's CV](https://m.canouil.dev/curriculum-vitae/curriculum-vitae.pdf)
```

> It feels a bit weird to me to use a html link inside an image markdown syntax. 

Well, the feature exists and Pandoc discourages the use of raw HTML when markdown-ish variants exist.

https://pandoc.org/MANUAL.html#philosophy

> while pandoc allows the embedding of raw HTML, it discourages it, [...]
